### PR TITLE
User story25

### PR DESCRIPTION
**Pull request by zachlp2016**
Added test that a visitor goes to the cart show page and sees a message if they haven't logged in or registered.
Added a test that the message has two links directing to the registration page and to the login page.

### CODE REVIEW

File: app/views/carts/show.html.erb [hunk 1]
```diff
@@ -1,3 +1,7 @@
+<% if current_user.nil? %>
+  <p>You must <%= link_to 'register', register_path %> or <%= link_to 'login', login_path %> to checkout.</p>
```
Comment (BrennanAyers) [inline app/views/carts/show.html.erb:2]: 👍

File: spec/features/carts/show_spec.rb [hunk 2]
```diff
@@ -16,18 +16,40 @@
...
     end
+
+    it 'shows all items in my cart and allows user to clear cart' do
+      merchant = create(:user, name: "Merchant", role: 1)
+      item_1 = create(:item, user: merchant)
+
+      visit items_path
+
+      within "#item-#{item_1.id}" do
+        click_button "Add To Cart"
```
Comment (BrennanAyers) [inline spec/features/carts/show_spec.rb]: @zachlp2016 Vince just changed these to links, do you mind changing this click_button to click_link?

Comment (zachlp2016) [reply spec/features/carts/show_spec.rb]: Awesome just changed it thanks Brennan!

Comment (BrennanAyers) [reply spec/features/carts/show_spec.rb]: Thanks!

**Review by BrennanAyers — APPROVED** [review]

**Pull request merged**